pallets/ajuna-affiliate/src/lib.rs

```diff
@@ -0,0 +1,272 @@
+	pub type Affiliators<T: Config> = StorageMap<_, Identity, T::AccountId, bool, ValueQuery>;
+
+	/// Stores the affiliated accounts from the perspective of the affiliator
+	#[pallet::storage]
+	pub type Affiliatables<T: Config> = StorageMap<_, Identity, T::AccountId, (), OptionQuery>;
+
+	/// Stores the affiliate logic rules
+	#[pallet::storage]
+	pub type AffiliateRules<T: Config> =
+		StorageMap<_, Blake2_128Concat, ExtrinsicId, u8, OptionQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// An organizer has been set.
+		OrganizerSet {
+			organizer: T::AccountId,
+		},
+		AccountAffiliated {
+			account: T::AccountId,
+			to: T::AccountId,
+		},
+		RuleAdded {
+			extrinsic_id: ExtrinsicId,
+		},
+		RuleCleared {
+			extrinsic_id: ExtrinsicId,
+		},
+		RuleExecuted {
+			extrinsic_id: ExtrinsicId,
+			account: T::AccountId,
+			beneficiary: T::AccountId,
+		},
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// There is no account set as the organizer
+		OrganizerNotSet,
+		/// An account cannot affiliate itself
+		CannotAffiliateSelf,
+		/// The account is not allowed to receive affiliates
+		CannotAffiliateToAccount,
+		/// This account has reached the affiliate limit
+		CannotAffiliateMoreAccounts,
+		/// This account has already been affiliated by another affiliator
+		CannotAffiliateAlreadyAffiliatedAccount,
+		/// This account is already an affiliator, so it cannot be affiliated by another account
+		CannotAffiliateExistingAffiliator,
+		/// The given extrinsic identifier is already paired with an affiliate rule
+		ExtrinsicAlreadyHasRule,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight({10_000})]
+		pub fn set_organizer(origin: OriginFor<T>, organizer: T::AccountId) -> DispatchResult {
+			ensure_root(origin)?;
+
+			Organizer::<T>::put(&organizer);
+			Self::deposit_event(Event::OrganizerSet { organizer });
+
+			Ok(())
+		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight({10_000})]
+		pub fn add_rule_to(
+			origin: OriginFor<T>,
+			extrinsic_id: ExtrinsicId,
+			rule: u8,
+		) -> DispatchResult {
+			let _ = Self::ensure_organizer(origin)?;
+
+			ensure!(
+				!AffiliateRules::<T>::contains_key(extrinsic_id),
+				Error::<T>::ExtrinsicAlreadyHasRule
+			);
+			AffiliateRules::<T>::insert(extrinsic_id, rule);
+			Self::deposit_event(Event::RuleAdded { extrinsic_id });
+
+			Ok(())
+		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight({10_000})]
+		pub fn clear_rule_from(origin: OriginFor<T>, extrinsic_id: ExtrinsicId) -> DispatchResult {
+			let _ = Self::ensure_organizer(origin)?;
+
+			AffiliateRules::<T>::remove(extrinsic_id);
+			Self::deposit_event(Event::RuleCleared { extrinsic_id });
+
+			Ok(())
+		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight({10_000})]
+		pub fn affiliate_to(origin: OriginFor<T>, affiliator: T::AccountId) -> DispatchResult {
+			let affiliatee = ensure_signed(origin)?;
```

Comment:
using the affiliatee name is inconsistent with  Affiliatees storage, we should make sure that the naming creates a certain consistency,

using child or parent or ... even more specific names.

just account for affiliatee.